framework/wazuh/core/cluster/cluster.py

```diff
@@ -250,50 +251,95 @@ def update_cluster_control_with_failed(failed_files, ko_files):
             ko_files['shared'].pop(f, None)
 
 
-def compress_files(name, list_path, cluster_control_json=None):
-    """Create a zip with cluster_control.json and the files listed in list_path.
+def compress_files(name, list_path, cluster_control_json):
+    """Create a compress file with cluster_control.json and the files listed in list_path.
 
-    Iterate the list of files and groups them in the zip. If a file does not
+    Iterate the list of files and groups them in a compressed file. If a file does not
     exist, the cluster_control_json dictionary is updated.
 
     Parameters
     ----------
     name : str
-        Name of the node to which the zip will be sent.
+        Name of the node to which the compress file will be sent.
     list_path : list
-        List of file paths to be zipped.
+        File paths to be zipped.
     cluster_control_json : dict
-        KO files (path-metadata) to be zipped as a json.
+        KO files (path-metadata) to be compressed as a json.
 
     Returns
     -------
-    zip_file_path : str
-        Path where the zip file has been saved.
+    compress_file_path : str
+        Path where the compress file has been saved.
     """
-    failed_files = list()
-    zip_file_path = path.join(common.wazuh_path, 'queue', 'cluster', name,
-                              f'{name}-{datetime.utcnow().timestamp()}-{str(random())[2:]}.zip')
-    if not path.exists(path.dirname(zip_file_path)):
-        mkdir_with_mode(path.dirname(zip_file_path))
-    with zipfile.ZipFile(zip_file_path, 'x') as zf:
-        # write files
-        if list_path:
-            for f in list_path:
+
+    def remove_from_cluster_control(file_to_remove):
+        """Discard file from 'missing' or 'shared' dicts inside cluster_control_json.
+
+        Parameters
+        ----------
+        file_to_remove : str
+            Relative file path, used as the key in the cluster_control_json dict.
+        """
+        try:
+            if not cluster_control_json['missing'].pop(file_to_remove, None):
+                cluster_control_json['shared'].pop(file_to_remove, None)
+        except (KeyError, TypeError):
+            pass
+
+    zip_size = 0
+    failed_files = []
+    exceeded_size = False
+    max_zip_size = get_cluster_items()['intervals']['communication']['max_zip_size']
+    compress_file_path = path.join(common.wazuh_path, 'queue', 'cluster', name,
+                                   f'{name}-{datetime.utcnow().timestamp()}-{str(random.random())[2:]}.compressed')
+
+    if not path.exists(path.dirname(compress_file_path)):
+        mkdir_with_mode(path.dirname(compress_file_path))
+
+    with open(compress_file_path, 'ab') as wf:
```

Comment:
In this function, we can reduce the level of indentation by inverting the order of the if and using continue